packages/timeline/src/browser/timeline-widget.tsx

```diff
@@ -0,0 +1,208 @@
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+
+/* eslint-disable no-null/no-null, @typescript-eslint/no-explicit-any */
+
+import { Message } from '@phosphor/messaging';
+import { inject, injectable, postConstruct } from 'inversify';
+import { DisposableCollection } from '@theia/core/lib/common/disposable';
+import {
+    ApplicationShell,
+    BaseWidget,
+    MessageLoop,
+    NavigatableWidget,
+    Panel,
+    PanelLayout,
+    Widget
+} from '@theia/core/lib/browser';
+import { TimelineTreeWidget } from './timeline-tree-widget';
+import { TimelineService } from './timeline-service';
+import { CommandRegistry } from '@theia/core/lib/common';
+import { TimelineEmptyWidget } from './timeline-empty-widget';
+import { toArray } from '@phosphor/algorithm';
+import URI from '@theia/core/lib/common/uri';
+import { TimelineProvidersChangeEvent } from '../common/timeline-model';
+import { TimelineAggregate } from './timeline-service';
+
+@injectable()
+export class TimelineWidget extends BaseWidget {
+
+    protected panel: Panel;
+    static ID = 'timeline-view';
+
+    private readonly timelinesBySource = new Map<string, TimelineAggregate>();
+
+    @inject(TimelineTreeWidget) protected readonly resourceWidget: TimelineTreeWidget;
+    @inject(TimelineService) protected readonly timelineService: TimelineService;
+    @inject(CommandRegistry) protected readonly commandRegistry: CommandRegistry;
+    @inject(ApplicationShell) protected readonly applicationShell: ApplicationShell;
+    @inject(TimelineEmptyWidget) protected readonly timelineEmptyWidget: TimelineEmptyWidget;
+
+    constructor() {
+        super();
+        this.id = TimelineWidget.ID;
+        this.addClass('theia-timeline');
+    }
+
+    @postConstruct()
+    protected init(): void {
+        const layout = new PanelLayout();
+        this.layout = layout;
+        this.panel = new Panel({
+            layout: new PanelLayout({
+            })
+        });
+        this.panel.node.tabIndex = -1;
+        layout.addWidget(this.panel);
+        this.containerLayout.addWidget(this.resourceWidget);
+        this.containerLayout.addWidget(this.timelineEmptyWidget);
+
+        this.refresh();
+        this.refreshList();
+        this.toDispose.push(this.timelineService.onDidChangeTimeline(event => {
+                const currentWidgetUri = this.getCurrentWidgetUri();
+                if (currentWidgetUri ) {
+                    this.loadTimeline(currentWidgetUri, event.reset);
+                }
+            })
+        );
+        this.toDispose.push(this.applicationShell.onDidChangeCurrentWidget(async event => {
+            const current = event.newValue;
+            if (current && NavigatableWidget.is(current)) {
```

Comment:
should not we do oppoositive if the current widget is no navigatable empty, i.e. clean up the timeline view

Reply:
We do it inside the next if condition:
https://github.com/eclipse-theia/theia/blob/ba77c86e3a3fd3513a123c7f1be11e17f93b1f0d/packages/timeline/src/browser/timeline-widget.tsx#L94-L97